be/src/vec/exec/format/orc/vorc_reader.cpp

```diff
@@ -253,13 +261,25 @@ Status OrcReader::_init_read_columns() {
         orc_cols_lower_case.emplace_back(_get_field_name_lower_case(&root_type, i));
     }
     for (auto& col_name : _column_names) {
+        if (_is_hive) {
+            auto iter = _scan_params.slot_name_to_schema_pos.find(col_name);
+            if (iter != _scan_params.slot_name_to_schema_pos.end()) {
+                int pos = iter->second;
+                orc_cols_lower_case[pos] = iter->first;
+            }
```

Comment:
What if `iter == end()`, can it happen? it not, may we should add a DCHECK or return error here